README.md

```diff
@@ -167,66 +193,82 @@ Because Javascript is single-threaded, forking does not actually cause the
 computation to be run in a separate thread. Forking just allows the subsequent
 actions to execute without waiting for the forked computation to complete.
 
-If the asynchronous computation supports it, you can "kill" a forked computation
-using the returned canceler:
+Forking returns a `Fiber eff a`, representing the deferred computation. You can
+kill a `Fiber` with `killFiber`, which will run any cancelers and cleanup, and
+you can observe a `Fiber`'s final value with `joinFiber`. If a `Fiber` threw
+an exception, it will be rethrown upon joining.
 
 ```purescript
-canceler <- forkAff myAff
-canceled <- canceler `cancel` (error "Just had to cancel")
-_        <- liftEff $ if canceled then (log "Canceled") else (log "Not Canceled")
+example = do
+  fiber <- forkAff myAff
+  killFiber (error "Just had to cancel") fiber
+  result <- try (joinFiber fiber)
+  if isLeft result
+    then (log "Canceled")
+    else (log "Not Canceled")
 ```
 
-If you want to run a custom canceler if some other asynchronous computation is
-cancelled, you can use the `cancelWith` combinator:
-
-```purescript
-otherAff `cancelWith` myCanceler
-```
 
 ## AVars
 
-The `Control.Monad.Aff.AVar` module contains asynchronous variables, which are very similar to Haskell's `MVar` construct. These can be used as low-level building blocks for asynchronous programs.
+The `Control.Monad.Aff.AVar` module contains asynchronous variables, which
+are very similar to Haskell's `MVar`. These can be used as low-level building
+blocks for asynchronous programs.
 
 ```purescript
-do v <- makeVar
-   _ <- forkAff do
-     delay (Milliseconds 50.0)
-     putVar v 1.0
-   a <- takeVar v
-   liftEff $ log ("Succeeded with " ++ show a)
+example = d
+  v <- makeEmptyVar
+  _ <- forkAff do
+    delay (Milliseconds 50.0)
+    putVar v 1.0
+  a <- takeVar v
+  log ("Succeeded with " <> show a)
 ```
```

Comment:
Is it worth briefly mentioning the take/put, etc. sematics `AVar` blocking here? If people aren't familiar with `MVar` it might not be obvious what the difference between this and a `Ref` is, even with the code example.